### ISSUE TITLE
How does Keras embedding work?

### ISSUE THREAD

**Issue by EpicWayne** [fchollet/keras#1269]
Hi, 
I'm trying to understand how the embedding layer in keras works. I assume that this performs word encodings, but I'm interested in how it is done. I have read the code, but it seems to use some gather function from theano which I cannot seem to find. Would someone be able to give me a better definition of what's going on? Thanks a bunch!

**Comment by fchollet** [comment]
The Embedding layer is really just a matrix, of shape (number_of_words, output_dimension). Each row in the matrix is the vector representation of one word. 

These vectors are learned together with the rest of the model via gradient descent. So they end up representing something meaningful wrt the loss function of your task.
